presto-native-execution/presto_cpp/main/operators/ShuffleInterface.h

```diff
@@ -43,46 +29,65 @@ class ShuffleInterface {
   /// @param success set to false to indicate aborted client.
   virtual void noMoreData(bool success) = 0;
 
+  /// Return true if all the data is finished writing and is ready to
```

Comment:
A question here: the writer calls a sequence of collect(), then call noMoreData() to signal the backend shuffle service, and then pool the write progress by readyForRead()? Is readyForRead() a blocking call or shall return a future which is fulfilled after all the write completes. Here finish writing means the write client flush all the data to the shuffle service?

Reply:
Good question. Let's discuss offline.